nagios.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+)
+
+// NagiosData is a task factory for tasks that fetch a JSON representation of
+// the provided project. The task uses outFor and errOutFor to get io.Writers
+// to write, respectively, the JSON output and any eventual error message.
+func NagiosData(project string, outFor, errOutFor projectResourceWriterCloserFactory) Task {
+	return nagiosData(func(project, pod string) *exec.Cmd {
+		return exec.Command("oc", "exec", pod, "cat", "/var/log/nagios/status.dat")
+	}, project, outFor, errOutFor)
+}
+
+// A getNagiosDataCmdFactory generates commands to get the raw nagios status data
+// in the provided project from the provided pod
+type getNagiosDataCmdFactory func(project, pod string) *exec.Cmd
+
+// nagiosData will dump a JSON representation of the current nagios status into
+// the io.Writer returned from outFor and log any errors it encounters doing so
+// into the io.Writer returned from errOutFor.
+func nagiosData(cmdFactory getNagiosDataCmdFactory, project string, outFor, errOutFor projectResourceWriterCloserFactory) Task {
+	return func() error {
+		stdout, stdoutCloser, err := outFor(project, "nagios")
+		if err != nil {
+			return err
+		}
+		defer stdoutCloser.Close()
+
+		stderr, stderrCloser, err := errOutFor(project, "nagios")
+		if err != nil {
+			return err
+		}
+		defer stderrCloser.Close()
+
+		var stdoutBuff bytes.Buffer
+
+		nagiosPod, err := getProjectPodNamesBySubstr(project, "nagios")
+		if err != nil {
+			fmt.Fprint(stderr, err)
+			return err
+		}
+
+		if len(nagiosPod) == 0 {
```

Comment:
> Whats nagiosPod here? Array or string?

To be precise it is neither of the two. It is a `[]string`, a.k.a "slice of string".

> @philbrookes and len(nagiosPod) == 0 is ugly. We need to change it to if(nagiosPods.isEmpty) cause in this context I dont know if that is string or array

There's no such a thing. @philbrookes's got it right, the Go way.

---

About the name, it is just a sign of a bigger dilemma. We "grep" something in search for the Nagios pod and we hope there is only one -- not zero, and not more.
The first part we're checking for, the second part we are just assuming that the first one is the one we care about (when we write below `nagiosPod[0]`).